data-processor/src/test/groovy/io/micronaut/data/processor/sql/ColumnTransformerSpec.groovy

```diff
@@ -153,6 +153,6 @@ class Project {
             def sql = builder.buildQuery(QueryModel.from(entity).eq("xyz", new QueryParameter("xyz"))).query
 
         expect:
-            sql == 'SELECT transform_."project_id_department_id",transform_."project_id_project_id",UPPER(xyz@abc) AS xyz FROM "transform" transform_ WHERE (UPPER(xyz@abc) = ?)'
+            sql == 'SELECT transform_."project_id_department_id",transform_."project_id_project_id",UPPER(xyz@abc) AS xyz FROM "transform" transform_ WHERE (transform_."xyz" = LOWER(xyz@abc))'
```

Comment:
This one doesn't look correct, doesn't even compare parameter with actual field value (or using transformation). Need to look more into it.

Reply:
Well, it's because it is weirdly defined transform that doesn't mean much in practice 
```
@ColumnTransformer(
            read = "UPPER(xyz@abc)",
            write = "LOWER(xyz@abc)"
    )
```
because `xyz@abc` doesn't represent anything. This is not a bug in the code change, just unexpected and inappropriate transform column configuration.

Reply:
Updated Transform class to have proper write expression using ? since using arbitrary field did not make much sense.